indigo_app/forms.py

```diff
@@ -952,10 +988,54 @@ def is_repealed_work_saved(self):
     def save(self):
         work = self.cleaned_data['repealed_work']
         if self.cleaned_data.get('DELETE'):
-            work.repealed_by = None
-            work.repealed_date = None
-            work.save()
+            if work.repealed_by == self.work:
+                work.repealed_by = None
+                work.repealed_date = None
+                work.save_with_revision(user=self.cleaned_data['updated_by_user'])
```

Comment:
I think you can pass self.request.user to form.save() from the view and use that instead — you could rename it as save_changes if this is a builtin that you're overriding, see WorkBulkUpdateForm. see https://github.com/laws-africa/indigo/commit/17fd3996f7a84b20f1ee07b602db3111d63ed085 (that's gone now so you won't see it elsewhere, but it worked! :))

Reply:
@goose-life We are calling repeals_made_form.save() from within the work form save method, I think its better to let the inner forms (repeals and amendments) manage their own data,  they can be responsible for setting and updating the user if they change. Since we don't have request object in the form class, the easiest way to pass the user data is as a form field.

Reply:
got it! okay I thought I might be missing something 😅